pkg/monitoring/migrations/prometheus/prometheus.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package prometheus
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const migrationNamespace = "kubevirt"
+const migrationSubsystem = "migration"
+
+type MigrationMetrics struct {
+	MigrationDuration *prometheus.HistogramVec
+}
+
+func NewMigrationMetrics() *MigrationMetrics {
+	return &MigrationMetrics{
+		MigrationDuration: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Namespace: migrationNamespace,
+				Subsystem: migrationSubsystem,
+				Name:      "duration_seconds",
+				Help:      "Time spent on migrations.",
+				// Will create 12 buckets, starting with migrations that take less then 2 seconds
+				// up to 2048 seconds(34,13 minutes), plus an extra for +Inf
+				Buckets: prometheus.ExponentialBuckets(2, 2, 11),
```

Comment:
I find the introduction of this histogram a little bit difficult. Migrations can take a long time. I am not sure what we consider to be too long to have a dedicated bucket. @sradco did you collect such metrics in oVirt and have some insight s which would help having a good first guess for meaningful bucket sizing?

Reply:
Yeah, I think the most appropriate approach would be to enable its configuration by the user, but that would require some API changes that I'm not familiar with yet

If you are using KubeVirt for small VMs, then you could set up small buckets... big buckets for big VMs, etc.

do you think we should introduce this metric with changes on the KubeVirt CR right from the start?

Reply:
Making it configurable from the beginning may indeed be the way to go. Let's see what @sradco thinks.

Reply:
@rmohr Sorry, I have no experience with buckets from oVirt.

Reply:
@sradco no problem, but maybe with typical migration runtimes.

Reply:
Sounds to me like we should go with configurable buckets ... :/

Let's add some nice defaults, something like 1m, 5m, 10m, 30m, 1h, 2h, 3h, 4h, 5h, 6h, 7h, 10h. I don't think that more granularity makes sense for defaults. @sradco @ArthurSens what do you think?

Reply:
I agree

Reply:
cool, I'll start working on it :slightly_smiling_face: